service/src/test/java/uk/gov/hmcts/reform/fpl/service/DraftCMOServiceTest.java

```diff
@@ -0,0 +1,125 @@
+        FormatStyle.MEDIUM).localizedBy(Locale.UK);
+
+    @Test
+    void shouldReturnHearingDateDynamicListWhenCaseDetailsHasHearingDate() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(ImmutableMap.of(
+                "hearingDetails", createHearingBookings(date))
+            )
+            .build();
+
+        DynamicList hearingList = draftCMOService.getHearingDateDynamicList(caseDetails);
+
+        assertThat(hearingList.getListItems())
+            .containsAll(Arrays.asList(
+                DynamicListElement.builder()
+                    .code("b15eb00f-e151-47f2-8e5f-374cc6fc2657")
+                    .label(date.plusDays(5).format(dateTimeFormatter))
+                    .build(),
+                DynamicListElement.builder()
+                    .code("6b3ee98f-acff-4b64-bb00-cc3db02a24b2")
+                    .label(date.plusDays(2).format(dateTimeFormatter))
+                    .build(),
+                DynamicListElement.builder()
+                    .code("ecac3668-8fa6-4ba0-8894-2114601a3e31")
+                    .label(date.format(dateTimeFormatter))
+                    .build()));
+    }
+
+    @Test
+    void shouldReturnHearingDateDynamicListWhenCmoHasPreviousSelectedValue() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(ImmutableMap.of(
+                "hearingDetails", createHearingBookings(date),
+                "caseManagementOrder", CaseManagementOrder.builder()
+                    .hearingDate(date.plusDays(2).format(dateTimeFormatter))
+                    .hearingDateId(fromString("6b3ee98f-acff-4b64-bb00-cc3db02a24b2").toString())
+                    .build()
+                )).build();
+
+        DynamicList hearingList = draftCMOService.getHearingDateDynamicList(caseDetails);
+
+        assertThat(hearingList.getListItems())
+            .contains(DynamicListElement.builder()
+                    .code("6b3ee98f-acff-4b64-bb00-cc3db02a24b2")
+                    .label(date.plusDays(2).format(dateTimeFormatter))
+                    .build());
+    }
+
+    @Test
+    void shouldReturnBlankHearingDateWhenPreviousSelectedValueNotMatchingDynamicListItems() {
+        final String expectedHearDateId = randomUUID().toString();
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(ImmutableMap.of(
+                "hearingDetails", createHearingBookings(date.plusDays(5)),
+                "caseManagementOrder", CaseManagementOrder.builder()
+                    .hearingDate(date.plusMonths(5).toString())
+                    .hearingDateId(expectedHearDateId)
+                    .build())
+            ).build();
+
+        DynamicList hearingList = draftCMOService.getHearingDateDynamicList(caseDetails);
+
+        assertThat(hearingList.getListItems())
+            .contains(DynamicListElement.builder()
+                .code(expectedHearDateId)
+                .label("")
+                .build());
+    }
```

Comment:
why is this ok to return? What if I select this value from the list?

Reply:
removed test as discussed.